### ISSUE TITLE
Gradle?

### ISSUE THREAD

**Issue by LemonLake** [itdelatrisu/opsu#125]
I personally think we should use gradle as a build system for the project. I'm not sure if it's natively supported in Eclipse, but IntelliJ support is definitely good. And, of course, maven repositories work with it.

I don't mind converting the project to gradle for this, this is more of a permission request.

**Comment by itdelatrisu** [comment]
* What are the advantages of this?  I've only used Gradle briefly to build some APKs, but I remember it being incredibly slow (I don't know how much of that was Gradle's fault, though).
* What are the changes this would involve?
* Would it be possible to add this as an additional build system instead of replacing the current one? (this probably depends on the previous answer)

**Comment by LemonLake** [comment]
The only change necessary would be to make a build.gradle file and scrap the old pom.xml file. Gradle seems to have a version of Maven's shade plugin. The advantages are incredibly cleaner build files that are easy to understand, easier jar dependencies and the ability to make tasks that can actually be comprehended. Using it as an additional build system would be possible, though adding the dependencies to the two is a little awkward.

**Comment by itdelatrisu** [comment]
If it wouldn't take too long, would you mind writing the Gradle build file first so I could try it before deciding?  Others might be able to give more input that way as well.  (I'm not particularly attached to Maven, but it's doing everything I want right now so I'll need some convincing to switch to something else. :P)

**Comment by LemonLake** [comment]
Yeah, sure. I'll give it a test-write now.

**Comment by LemonLake** [comment]
I haven't yet added natives or jarsplice due to my lack of understanding of pom files, but natives are something I know how to do and will do after this post. Before my update on this, here's how build.gradle looks:

```
apply plugin: 'java'
apply plugin: 'maven'

group = 'itdelatrisu'
version = '0.10.1'

sourceCompatibility = 1.7
targetCompatibility = 1.7

apply plugin: 'shadow'

repositories {
    maven { url "http://repo.maven.apache.org/maven2" }
}

dependencies {
    compile 'org.lwjgl.lwjgl:lwjgl:2.9.3'
    compile 'org.slick2d:slick2d-core:1.0.1'
    compile 'org.jcraft:jorbis:0.0.17'
    compile 'net.lingala.zip4j:zip4j:1.3.2'
    compile 'com.googlecode.soundlibs:jlayer:1.0.1-1'
    compile 'com.googlecode.soundlibs:mp3spi:1.9.5-1'
    compile 'com.googlecode.soundlibs:tritonus-share:0.3.7-2'
    compile 'org.xerial:sqlite-jdbc:3.8.6'
    compile 'org.json:json:20140107'
    compile 'net.java.dev.jna:jna:4.1.0'
    compile 'net.java.dev.jna:jna-platform:4.1.0'
    compile 'org.apache.maven:maven-artifact:3.3.3'
    compile 'org.apache.commons:commons-compress:1.9'
    compile 'org.tukaani:xz:1.5'
    compile 'com.github.jponge:lzma-java:1.3'
    compile fileTree(include: ['*.jar'], dir: 'lib')
}
```

I'll post back here when I've added natives and jarsplice.

**Comment by LemonLake** [comment]
Addressed in #126.
